app.rb

```diff
@@ -42,15 +42,31 @@ class App < Sinatra::Base
 
       begin
         analysis = Exercism::Analysis.new(config.adapter.new(code)).run(*config.analyzers)
-        results = analysis.values.reject {|result|
-          result.feedback.empty?
-        }.map {|result|
-          {type: result.type, keys: result.feedback.map(&:type).uniq}
-        }
+
+        results = reject_invalid_results(code, analysis.values)
+
+        results = results.map do |result|
+          { type: result.type, keys: result.feedback.map(&:type).uniq }
+        end
       rescue => e
         halt 500, {error: e.message + "\n" + e.backtrace[0] }.to_json
       end
       halt 200, {results: results}.to_json
     end
+
+    def code_from_test_file?(code)
```

Comment:
It's a good idea to order your methods so that they are encountered in the order they are used reading from top to bottom. In this case `code_from_test_file?` has not been referenced yet. It would be better placed under the `reject_invalid_results` method.